### PR TITLE
🔖 Update civic liker monthly report

### PR DESCRIPTION
**Pull request by shellyYG**

### CODE REVIEW

File: src/components/avatar.tsx [hunk 1]
```diff
@@ -62,3 +62,49 @@ export const Avatar = (props: AvatarProps) => {
     </MjmlRaw>
   );
 };
+
+export const SmallAvatar = (props: AvatarProps) => {
```
Comment (nwingt) [inline src/components/avatar.tsx]: I think this one can combine with Avatar? What is the effect of unwrapping `<MjmlRaw />`

File: src/components/classic-badge.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                  display: 'block',
+                  width: '100%',
+                  margin: '0 auto',
+                  maxWidth: MAX_WIDTH,
+                }}
+                alt=""
+              />
+            </td>
+          </tr>
+        </MjmlTable>
```
Comment (nwingt) [inline src/components/classic-badge.tsx]: How about replacing with MjmlImage ?

File: src/templates/reports/monthly/civic-liker/v2.types.ts [hunk 3]
```diff
@@ -12,6 +15,8 @@ export interface MonthlyReportCivicLikerV2TemplateProps extends MonthlyReport {
   totalSupportedCreatorsByFund?: number;
   billingDateTimestamp?: number;
   subscribingCreators?: SubscribingCreator[];
+  creatorfundingCreators?: CreatorfundingCreator[];
```
Comment (nwingt) [inline src/templates/reports/monthly/civic-liker/v2.types.ts]:
```suggestion
  fundedCreators?: FundedCreator[];
```

File: test/stub/civic-liker.ts [hunk 4]
```diff
@@ -44,6 +44,13 @@ export const MonthlyReportCivicLikerV2SampleData: MonthlyReportCivicLikerV2Templ
     isCivicLiker: true,
     amount: 5,
   }),
+  creatorfundingCreators: new Array(25).fill({
```
Comment (nwingt) [inline test/stub/civic-liker.ts]:
```suggestion
  likedCreators: new Array(25).fill({
```

File: src/templates/reports/monthly/civic-liker/v2.types.ts [hunk 5]
```diff
@@ -3,6 +3,9 @@ import { Content, MonthlyReport, User } from '../../../../types';
 export interface SubscribingCreator extends User {
   amount?: number;
 }
+export interface CreatorfundingCreator extends User {
```
Comment (nwingt) [inline src/templates/reports/monthly/civic-liker/v2.types.ts]:
```suggestion
export interface FundedCreator extends User {
```

File: src/templates/reports/monthly/civic-liker/v2.tsx [hunk 6]
```diff
@@ -142,114 +175,99 @@ export const MonthlyReportCivicLikerV2Template = (
...
-        <BasicSection paddingTop={0}>
-          <MjmlColumn>
+      <BasicSection paddingTop={0} paddingBottom={24} backgroundColor="white">
+        <MjmlGroup>
+          <TextColumn
+            messageId={
+              'report.monthly.civic-liker.creatorsfund.badge.description.start'
+            }
+            width="40%"
+          />
```
Comment (nwingt) [inline src/templates/reports/monthly/civic-liker/v2.tsx]: I think this use case is not necessary to turn into a component?

File: src/templates/reports/monthly/civic-liker/v2.tsx [hunk 7]
```diff
@@ -60,8 +93,8 @@ export const MonthlyReportCivicLikerV2Template = (
     totalSupportedCreatorsByFund = 0,
     billingDateTimestamp,
     subscribingCreators = [],
-    supportedContents = [],
-    supportedContentsRemainsCount = 0,
+    creatorfundingCreators = [],
```
Comment (nwingt) [inline src/templates/reports/monthly/civic-liker/v2.tsx]:
```suggestion
    fundedCreators = [],
```

File: src/templates/reports/monthly/civic-liker/v2.tsx [hunk 8]
```diff
@@ -45,6 +54,30 @@ const MonthlyReportCivicLikerV2SubscribersDescription = ({
   );
 };
 
+const MonthlyReportCivicLikerV2CreatorsFundDescription = ({
+  creatorfundingCreators,
+}: {
+  creatorfundingCreators: Array<any>;
```
Comment (nwingt) [inline src/templates/reports/monthly/civic-liker/v2.tsx]: simply given the number is enough? Seem you don't use the array but only the length

File: src/i18n/translations/cn.json [hunk 9]
```diff
@@ -18,13 +18,20 @@
...
   "report.monthly.civic-liker.supported.creators": "创作者",
   "report.monthly.civic-liker.supported.v1": "你支持了",
   "report.monthly.civic-liker.supported.v2": "已支持作者人数",
+  "report.monthly.civic-liker.creatorsfund.title": "创作基金",
+  "report.monthly.civic-liker.creatorsfund.description.main": "高达 10,000 美元的额外创作基金，将透过赞赏公民的按赞而分配给作者。<a>{count} 位作者因你的按赞而获得了额外奖励!</a>",
+  "report.monthly.civic-liker.creatorsfund.description.none": "高达 10,000 美元的额外创作基金，将透过赞赏公民的按赞而分配给作者。<a>一起按赞文章，支持作者！</a>",
+  "report.monthly.civic-liker.creatorsfund.contents.more": "还有 {count} 个作者",
+  "report.monthly.civic-liker.creatorsfund.contents.none": "点击LikeCoin赞赏键，支持你喜欢的创作者",
+  "report.monthly.civic-liker.creatorsfund.badge.description.start": "你也按赞支持了",
+  "report.monthly.civic-liker.creatorsfund.badge.description.end": "位创作者",
```
Comment (nwingt) [inline src/i18n/translations/cn.json]:
```suggestion
  "report.monthly.civic-liker.creatorsfund.badge.count.prepend": "你也按赞支持了",
  "report.monthly.civic-liker.creatorsfund.badge.count.append": "位创作者",
```

**Review by nwingt — CHANGES_REQUESTED** [review]

File: src/templates/reports/monthly/civic-liker/v2.tsx [hunk 10]
```diff
@@ -142,114 +166,101 @@ export const MonthlyReportCivicLikerV2Template = (
...
+                        key={creator.likerID}
+                        src={creator.avatarSrc}
+                        likerID={creator.likerID}
+                        displayName={creator.displayName}
+                        isCivicLiker={creator.isCivicLiker}
+                        size={36}
+                      />
+                    )
+                  );
+                })}
```
Comment (nwingt) [inline src/templates/reports/monthly/civic-liker/v2.tsx]: I do think using `.slice` is more decriptive
```suggestion
              {creatorfundingCreators &&
                creatorfundingCreators.slice(0, 11).map((creator, index) => (
                  <SmallAvatar
                    key={creator.likerID}
                    src={creator.avatarSrc}
                    likerID={creator.likerID}
                    displayName={creator.displayName}
                    isCivicLiker={creator.isCivicLiker}
                    size={36}
                  />
                )
              )}
```

**Review by nwingt — CHANGES_REQUESTED** [review]

File: src/i18n/translations/cn.json [hunk 11]
```diff
@@ -18,13 +18,20 @@
...
+  "report.monthly.civic-liker.supported.contents.more": "还有 {count} 作品",
   "report.monthly.civic-liker.supported.creators.fund": "按赞分配创作基金",
   "report.monthly.civic-liker.supported.creators.subscription": "赞赏公民订阅",
   "report.monthly.civic-liker.supported.creators.title": "首几位你支持了的作者",
   "report.monthly.civic-liker.supported.creators": "创作者",
   "report.monthly.civic-liker.supported.v1": "你支持了",
   "report.monthly.civic-liker.supported.v2": "已支持作者人数",
+  "report.monthly.civic-liker.creatorsfund.title": "创作基金",
+  "report.monthly.civic-liker.creatorsfund.description.main": "高达 10,000 美元的额外创作基金，将透过赞赏公民的按赞而分配给作者。<a>{count} 位作者因你的按赞而获得了额外创作基金奖励!</a>",
+  "report.monthly.civic-liker.creatorsfund.description.none": "高达 10,000 美元的额外创作基金，将透过赞赏公民的按赞而分配给作者。<a>一起按赞文章，支持作者！</a>",
```
Comment (shellyYG) [inline src/i18n/translations/cn.json:31]: test adding suggestion

File: src/components/header.tsx [hunk 12]
```diff
@@ -13,13 +16,52 @@ export const HeaderSection = () => {
...
+            color={Colors.Grey4A}
+            fontSize={14}
+            fontWeight={600}
+            lineHeight={'3.0'}
+            align="right"
+          >
+            <FormattedMessage id="download" />
+          </MjmlText>
+        </MjmlColumn>
+        <MjmlColumn width={'10%'} cssClass="hide_on_mobile">
```
Comment (nwingt) [inline src/components/header.tsx]:
```suggestion
        <MjmlColumn width="10%" cssClass="hide_on_mobile">
```

File: src/components/header.tsx [hunk 13]
```diff
@@ -13,13 +16,52 @@ export const HeaderSection = () => {
...
+              target="_blank"
+              rel="noreferrer noopener"
+              color={Colors.Grey9B}
+            >
+              iOS
+            </a>
+          </MjmlText>
+        </MjmlColumn>
+        <MjmlColumn width={'10%'} cssClass="hide_on_mobile">
+          <MjmlText fontSize={14} lineHeight={'3.0'}>
```
Comment (nwingt) [inline src/components/header.tsx]:
```suggestion
        <MjmlColumn width="10%" cssClass="hide_on_mobile">
          <MjmlText fontSize={14} lineHeight="3.0">
```

File: src/components/avatar.tsx [hunk 14]
```diff
@@ -62,3 +64,20 @@ export const Avatar = (props: AvatarProps) => {
     </MjmlRaw>
   );
 };
+
+export const CivicLiker = (props: AvatarProps) => {
```
Comment (nwingt) [inline src/components/avatar.tsx]: Why do u need another component for CivicLiker Icon？

File: src/components/classic-badge.tsx [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+  MjmlSectionProps,
+  MjmlImage,
+  PaddingProps,
+} from 'mjml-react';
+
+import { getAssetPath } from '../utils/url';
+
+import { BasicSection, BasicSectionProps } from './sections/basic';
+
+export const ClassicBadge = (
```
Comment (nwingt) [inline src/components/classic-badge.tsx]:
```suggestion
export const CivicLikerClassicBadge = (
```

File: src/templates/reports/monthly/civic-liker/v2.tsx [hunk 16]
```diff
@@ -142,114 +166,100 @@ export const MonthlyReportCivicLikerV2Template = (
...
+      </BasicSection>
+
+      <BasicSection paddingBottom={24} backgroundColor="white">
+        <MjmlColumn>
+          <MjmlText
+            fontSize={42}
+            fontWeight={600}
+            align="center"
+            color={Colors.Grey4A}
+            lineHeight={'0.5'}
```
Comment (nwingt) [inline src/templates/reports/monthly/civic-liker/v2.tsx]:
```suggestion
            lineHeight="0.5"
```

**Review by nwingt — CHANGES_REQUESTED** [review]

File: src/components/template-base.tsx [hunk 17]
```diff
@@ -55,6 +56,17 @@ export const TemplateBase = (props: TemplateBaseProps) => {
...
+            <MjmlStyle>
+              {`
+                  @media only screen and (max-width: 480px) { 
+                    *[class~=hide_on_mobile] { display: none !important;} 
+                    *[class~=header_columns_left] { width: 45% !important;}
+                    *[class~=header_columns_right] { width: 55% !important;}
+                    *[class~=hoverTable] { table-layout: auto !important;}
+                    
+                  }
+                `}
```
Comment (nwingt) [inline src/components/template-base.tsx]: Please fix the indentation

File: src/components/sections/monthly-report-header.tsx [hunk 18]
```diff
@@ -1,16 +1,24 @@
...
+  MjmlGroup,
+  MjmlText,
+  MjmlWrapper,
+  MjmlTable,
+} from 'mjml-react';
 import { FormattedMessage, useIntl } from 'react-intl';
-
 import * as Colors from '../../constants/colors';
-
+import { createCanvas } from 'canvas';
```
Comment (nwingt) [inline src/components/sections/monthly-report-header.tsx:11]: Please move this import under react-intl

File: src/components/avatar.tsx (hunk 14 above)
Comment (shellyYG) [reply src/components/avatar.tsx]: true. removed!

**Review by nwingt — CHANGES_REQUESTED** [review]

File: src/templates/reports/monthly/civic-liker/v2.tsx [hunk 19]
```diff
@@ -94,162 +133,167 @@ export const MonthlyReportCivicLikerV2Template = (
...
+          )}
+          {subscribingCreators.length > 0 && (
+            <MjmlText paddingBottom={16} fontSize={16}>
+              <MonthlyReportCivicLikerV2SubscribersDescription
+                language={language}
+                billingDateTimestamp={billingDateTimestamp}
+              />
+            </MjmlText>
+          )}
+          {subscribingCreators.length > 0 && (
```
Comment (nwingt) [inline src/templates/reports/monthly/civic-liker/v2.tsx]: Can refactor `subscribingCreators.length > 0` as `hasSubscribingCreators`

File: src/components/header.tsx [hunk 20]
```diff
@@ -13,13 +16,52 @@ export const HeaderSection = () => {
...
+            <a
+              href={`${LIKER_LAND_ROOT}/getapp`}
+              target="_blank"
+              rel="noreferrer noopener"
+              color={Colors.Grey9B}
+            >
+              Android
+            </a>
+          </MjmlText>
+        </MjmlColumn>
```
Comment (nwingt) [inline src/components/header.tsx]: Please refactor the `<a/> element`

**Review by nwingt — CHANGES_REQUESTED** [review]
Please also provide Civic Liker Classic samples in `examples/index.html`

File: src/components/link.tsx [hunk 21]
```diff
@@ -40,3 +45,17 @@ export const Link = (props: LinkProps) => {
...
+  const { operatingSystem } = props;
+  return (
+    <a
+      href={`${LIKER_LAND_ROOT}/getapp`}
+      target="_blank"
+      rel="noreferrer noopener"
+      color={Colors.Grey9B}
+    >
+      {operatingSystem}
+    </a>
```
Comment (nwingt) [inline src/components/link.tsx]: Can use React standard props `children`

**Review by nwingt — CHANGES_REQUESTED** [review]

**Review by nwingt — APPROVED** [review]

**Pull request merged**